packages/blockchain-link/README.md

```diff
@@ -1,55 +1,83 @@
+
+All methods can fail and return an error eg. in case the BE is offline.
+
+- `getInfo`: Get general information about the network, like current block-height, name, smallest possible division, etc.
+- `getBlockHash`: Get hash of a block of given height.
+- `getAccountInfo`: Get info about an account, eg. derived addresses, balance, transaction history etc.
+- `getAccountUtxo`: Get unspent inputs for given account. Only bitcoin(like) networks.
+- `getAccountBalanceHistory`: Get historical progression of given account's balance. Used for rendering a graph in Suite's dashboard.
+- `getCurrentFiatRates`: Get current fiat rates.
+- `getFiatRatesForTimestamps`: Get historical fiat rates, only some networks support this.
+- `getFiatRatesTickersList`: Get fiat currencies for which rates are available.
+- `estimateFee`: Get ‘recommended’ fee value inferred from current traffic in the network.
+- `subscribe`: Subscribe for live changes in
+    - blockchain i.e new blocks mined.
+    - accounts, addresses i.e. new transactions broadcasted or mined.
+    - fiatRates
+    - connection to BE (HANDSHAKE, CONNECT, DISCONNECT)
+    
+    Handling subscription state is left to the user. (Hence the purpose of HANDSHAKE, CONNECT and DISCONNECT notifications.)
+- `unsubscribe`: Cancel existing subscription.
+- `getTransaction`: Get info about a given transaction. Return value of this method is not identical across networks; this method exposes the specific transaction format.
+- `pushTransaction`: Broadcast given transaction to the network.
+
+## Usage
+
+Add blockchain-link to your dependencies.
+
+```shell
+yarn add @trezor/blockchain-link
+```
 
--   [blockbook](https://github.com/trezor/blockbook/)
--   [ripple-lib](https://github.com/ripple/ripple-lib/)
+And use it.
 
-## Development:
+```javascript
+import BlockchainLink from '@trezor/blockchain-link';
 
-Compile and run with simple ui for the API methods.
+const link = new BlockchainLink({
+    name: 'Name used in logs.';
+    worker: 'path/to/the/worker.js';
+    server: ['url1.of.the.be', 'url2.of.the.be'];
+    debug: true;
+});
 
+try {
+    const resp = link.getInfo();
+} catch(error) {
+
+}
 ```
+
+## Development
+
+This package provides a simple testing UI for playing around with various implementations and BEs. Run it with
+
+```shell
 yarn
 yarn dev
 ```
 
-## Build
+### Build
 
 ```
 yarn lint
 yarn test
 yarn build
 ```
 
-## Integration
-
-```
-yarn add @trezor/blockchain-link
-```
-
-```
-import BlockchainLink from '@trezor/blockchain-link';
-
-const link = new BlockchainLink({
-    name: string;
-    worker: string;
-    server: string[];
-    debug: boolean;
-});
-
-try {
-    const resp = link.getInfo();
-} catch(error) {
-
-}
-```
+### Workers compilation
 
-## Workers compilation
+Workers are already built and minified inside `@trezor/blockchain-link/build/` directory.
```

Comment:
`./build/module` - contains files where worker is wrapped by primitive self written [webpack plugin ](https://github.com/trezor/trezor-suite/blob/develop/packages/blockchain-link/webpack/module-worker-loader.j) into javascript module which can be used as regular module on the main thread. This build is useful for environments where `Worker` is not available, like `react-native`. see [webpack config](https://github.com/trezor/trezor-suite/blob/develop/packages/blockchain-link/webpack/workers.module.babel.js).

`./build/web` - contains files for browser based environments. see [webpack config](https://github.com/trezor/trezor-suite/blob/develop/packages/blockchain-link/webpack/workers.web.babel.js).

`./build/node` - contains files for node based environments. see [webpack config](https://github.com/trezor/trezor-suite/blob/develop/packages/blockchain-link/webpack/workers.node.babel.js).

All workers could be also build from source on the implementation-side. For example using webpack `worker-loader`:
```
import BlockbookWorker from 'worker-loader?filename=workers/blockbook-worker.[hash].js!@trezor/blockchain-link/lib/workers/blockbook/index.js';
```

Reply:
Thanks. Incorporated in fb7d3190c5d24e3924ce8f44394b175abcb413c2